src/reducers/ui.js

```diff
@@ -255,6 +261,21 @@ export default (state = EMPTY_UI, action) => {
                 showDetailsPanel: action.value ? false : state.showDetailsPanel,
             }
         }
+        case TOGGLE_UI_SIDEBAR_HIDDEN: {
+            return { ...state, hideMainSideBar: !state.hideMainSideBar }
+        }
+        case TOGGLE_UI_LAYOUT_PANEL_HIDDEN: {
+            return { ...state, hideLayoutPanel: !state.hideLayoutPanel }
+        }
+        case TOGGLE_UI_EXPANDED_VISUALIZATION_CANVAS: {
+            // Toggle both to `true` unless both already are
+            const nextValue = !(state.hideMainSideBar && state.hideLayoutPanel)
```

Comment:
It seems odd to call a selector from a reducer, so I've written it like above. However, technically the line above and `sGetUiShowExpandedVisualizationCanvas` contain duplicate logic, so I could also rewrite it to:
```suggestion
            const nextValue = !sGetUiShowExpandedVisualizationCanvas(state)
```